crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -52,8 +59,25 @@ impl<Provider, Pool> EthFilter<Provider, Pool> {
             max_logs_per_response,
             eth_cache,
             max_headers_range: MAX_HEADERS_RANGE,
-            task_spawner,
+            task_spawner: task_spawner.clone(),
         };
+
+        let active_filters_arc = Arc::clone(&inner.active_filters.inner);
+
+        inner.task_spawner.spawn_critical(
+            "stale-filters-cleaner",
+            Box::pin(async move {
+                let mut interval = tokio::time::interval(stale_filter_ttl);
+                loop {
+                    interval.tick().await;
+                    {
+                        active_filters_arc.lock().await.retain(|_, filter| {
+                            filter.last_poll_timestamp.elapsed() < stale_filter_ttl
```

Comment:
let's make this an async function of `EthFilter` instead, and move the `stale_filter_ttl` to a field,

so the service task only needs to call a function

Reply:
Ok! Done!